io/pci/distro_tools.py

```diff
@@ -30,27 +30,61 @@ class DisrtoTool(Test):
     '''
     to test different type of tool
     '''
-    @skipUnless(IS_POWER_VM,
-                "supported only on PowerVM platform")
+
     def setUp(self):
         '''
-        to check host platform
+        get all parameters
         '''
-        self.option = self.params.get("option", default='')
+        self.option = self.params.get("test_opt", default='').split(",")
+        self.tool = self.params.get("tool", default='')
 
-    def test(self):
+    @skipUnless(IS_POWER_VM,
+                "supported only on PowerVM platform")
+    def lsslot(self, option):
         '''
-        test the lsslot tool
+        run lsslot
         '''
         cmd = "lsslot"
-        if self.option == "pci":
-            cmd = "%s -d %s" % (cmd, self.option)
+        if option == "pci":
+            cmd = "%s -d %s" % (cmd, option)
         else:
-            cmd = "%s -c %s" % (cmd, self.option)
+            cmd = "%s -c %s" % (cmd, option)
         result = process.run(cmd, shell=True, ignore_status=True)
         if result.exit_status != 0:
             self.fail("Failed to display hot plug slots")
 
+    def netstat(self, option):
+        '''
+        run netstat
+        '''
+        cmd = "netstat -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("Failed to monitoring network connections")
+
+    def lsprop(self, option):
+        '''
+        run lsprop
+        '''
+        cmd = "lsprop -%s" % option
+        result = process.run(cmd, shell=True, ignore_status=True)
+        if result.exit_status != 0:
+            self.fail("lsprop failed")
+
+    def test(self):
+        '''
+        test different distro tools
+        '''
+        if self.tool == "lsslot":
+            for val in self.option:
+                self.lsslot(val)
+        elif self.tool == "netstat":
+            for val in self.option:
+                self.netstat(val)
+        elif self.tool == "lsprop":
+            for val in self.option:
+                self.lsprop(val)
+
```

Comment:
Test fails if any one of the tool or tool option fails.
failing/erroing out the test after checking all the tools along with corresponding all option would be ideal.

Reply:
this is not resolved right?
self.err = []
if result.exit_status != 0:
    self.err.append(option)
--->> add options too if possible here
and at the end of script, please check
if self.err:
    self.fail("few tools failed %s " % self.err)